cpp/src/copying/concatenate.cu

```diff
@@ -251,15 +247,16 @@ fused_concatenate_kernel(
   while (output_index < output_size) {
 
     // Lookup input index by searching for output index in offsets
-    auto const offset_it = thrust::upper_bound(thrust::seq,
-        input_offsets, input_offsets + input_size, output_index) - 1;
-    size_type const input_index = offset_it - input_offsets;
+    auto const offset_it =
+        thrust::upper_bound(thrust::seq, input_offsets,
+                            input_offsets + num_input_views, output_index) - 1;
+    size_type const partition_index = offset_it - input_offsets;
```

Comment:
I'd do this instead of the `- 1` above, it's more explicit and easier to notice. 
```suggestion
    size_type const partition_index = thrust::prev(offset_it) - input_offsets;
```